clients/rest-client-core/src/test/java/uk/gov/justice/services/clients/core/RestClientProcessorIT.java

```diff
@@ -173,6 +173,28 @@ public void shouldThrowExceptionOnQueryParamMissing() throws Exception {
         restClientProcessor.request(endpointDefinition, requestEnvelopeParamAParamB());
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowExceptionOnCPPIDMissing() throws Exception {
+
+        final String path = "/my/resource";
+        final String mimetype = format("application/vnd.%s+json", CONTEXT_QUERY_MY_QUERY);
+
+        stubFor(get(urlPathEqualTo(path))
+                .withHeader("Accept", WireMock.equalTo(mimetype))
+                .withQueryParam("paramA", WireMock.equalTo("valueA"))
+                .withQueryParam("paramC", WireMock.equalTo("valueC"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", mimetype)
+                        .withBody(envelopeWithoutMetadataAsJson)));
+
+        Set<QueryParam> queryParams = ImmutableSet.of(new QueryParam("paramA", true), new QueryParam("paramB", false), new QueryParam("paramC", true));
+
+        EndpointDefinition endpointDefinition = new EndpointDefinition(BASE_URI, path, emptySet(), queryParams);
+
+        validateResponse(restClientProcessor.request(endpointDefinition, requestEnvelopeParamAParamC()), envelopeWithMetadataAsJson);
```

Comment:
I don't think we need to call validateResponse here.  restClientProcessor.request will throw the exception.